pyanaconda/modules/localization/installation.py

```diff
@@ -98,80 +98,86 @@ def name(self):
         return "Configure keyboard"
 
     def run(self):
-        if self._x_layouts:
-            write_x_configuration(
+        localed_wrapper = LocaledWrapper()
+
+        x_layouts = self._x_layouts
+        vc_keymap = self._vc_keymap
+
+        if not self._x_layouts or not self._vc_keymap:
+            x_layouts, vc_keymap = get_missing_keyboard_configuration(
+                localed_wrapper,
                 self._x_layouts,
+                self._vc_keymap
+            )
+
+        if x_layouts:
+            write_x_configuration(
+                localed_wrapper,
+                x_layouts,
                 self._switch_options,
+                X_CONF_DIR,
                 self._sysroot
             )
-        if self._vc_keymap:
+        if vc_keymap:
             write_vc_configuration(
-                self._vc_keymap,
+                vc_keymap,
                 self._sysroot
             )
 
 
-def write_x_configuration(x_layouts, switch_options, root):
+def write_x_configuration(localed_wrapper, x_layouts, switch_options, x_conf_dir_path, root):
```

Comment:
I'm not familiar with this, so I am not sure why the `localed_wrapper` instance is passed from the caller. It seems redundant to pass when it wraps the same thing and you could just construct a new one, as you did? I can see how it makes tests easier though...

Reply:
The tests are the reason here. I am even considering passing the wrapper instance to the task(s) (to be shared by the module).

Reply:
Thank you, noted.